fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/FluidVariantPropertyHandler.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.fluid;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.sound.SoundEvents;
+import net.minecraft.text.Text;
+import net.minecraft.world.World;
+
+/**
+ * Defines the common properties of {@linkplain FluidVariant fluid variants} of a given Fluid.
+ * Register with {@link FluidVariantProperties#register}.
+ *
+ * <p><b>Experimental feature</b>, we reserve the right to remove or change it without further notice.
+ * The transfer API is a complex addition, and we want to be able to correct possible design mistakes.
+ */
+@ApiStatus.Experimental
+public interface FluidVariantPropertyHandler {
```

Comment:
"Property handler" sounds something much more generic and more active (think "event handler"). I'd call this `FluidVariantProperties` if it wasn't already in use for the util class.

Reply:
It's meant to mirror the render handler. Technically it can perform arbitrary logic based on the NBT of the variant, so it is a handler somewhat.

Reply:
Maybe also note that `RuntimeException` (or whatever exception you later come up with) will be thrown if the handler returns invalid values

Reply:
I downgraded the exception to a `LOGGER.warn` since it's not worth crashing the game for this.